### PR TITLE
Log errors returned from graphql

### PR DESCRIPTION
**Pull request by WithoutPants**
Adds an error presenter to the graphql interface that logs any encountered errors to the error log level. Also logs the request arguments at the debug log level when an error is encountered.

### CODE REVIEW

**Comment by WithoutPants** [comment]
Added logging for when the user logs in or out, when invalid credentials are entered, and when credentials are changed.

File: pkg/session/session.go [hunk 1]
```diff
@@ -34,7 +34,15 @@ const (
 	passwordFormKey = "password"
 )
 
-var ErrInvalidCredentials = errors.New("invalid username or password")
+type InvalidCredentialsError struct {
+	Username string
```
Comment (bnkai) [inline pkg/session/session.go:38]: Is this actually used anywhere? I only see it set when initializng ?
Wouldnt an empty struct be enough?

File: internal/api/error.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		s, _ := json.Marshal(args)
+		if len(s) > 0 {
+			args = string(s)
+		}
+
+		return "%s: %v", []interface{}{
+			graphql.GetFieldContext(ctx).Path(),
+			args,
+		}
+	})
```
Comment (bnkai) [inline internal/api/error.go:31]: An extra check is needed here `if graphql.GetFieldContext(ctx) != nil`
If the graphql query is mispelled/missing  then `GetFieldContext` returns nil and thus we get a panic
To repro try running any mispelled or non exitsing query/mutation

File: pkg/session/session.go (hunk 1 above)
Comment (WithoutPants) [reply pkg/session/session.go:38]: I originally had this field included in the `Error` method, but I've taken it out for now (until we support multiple users). Having the field means that we can read the username from the calling code, so it's useful, just not currently in use.

**Review by bnkai — APPROVED** [review]

**Pull request merged**